NWebDav.Server/Stores/DiskStoreItem.cs

```diff
@@ -15,38 +15,52 @@ public sealed class DiskStoreItem : IStoreItem
     private readonly DiskStoreBase _store;
     private readonly ILogger<DiskStoreItem> _logger;
 
+    /// <inheritdoc/>
+    public string Name { get; }
+
+    /// <inheritdoc/>
+    public string UniqueKey { get; }
+
+    /// <inheritdoc/>
+    public IPropertyManager PropertyManager { get; }
+
     public DiskStoreItem(DiskStoreBase store, DiskStoreItemPropertyManager propertyManager, FileInfo fileInfo, ILogger<DiskStoreItem> logger)
     {
         _store = store;
-        FileInfo = fileInfo;
-        PropertyManager = propertyManager;
         _logger = logger;
+        Name = fileInfo.Name;
+        UniqueKey = fileInfo.FullName;
+        PropertyManager = propertyManager;
+        FileInfo = fileInfo;
     }
 
-    public IPropertyManager PropertyManager { get; }
-
     public FileInfo FileInfo { get; }
     public bool IsWritable => _store.IsWritable;
-    public string Name => FileInfo.Name;
-    public string UniqueKey => FileInfo.FullName;
-    public string FullPath => FileInfo.FullName;
-    public Task<Stream> GetReadableStreamAsync(CancellationToken cancellationToken) => Task.FromResult((Stream)FileInfo.OpenRead());
 
+    /// <inheritdoc/>
+    public async Task<Stream> GetReadableStreamAsync(CancellationToken cancellationToken)
+    {
+        await Task.CompletedTask;
+        return FileInfo.OpenRead();
+    }
```

Comment:
This is not ok. If you're already awaiting, then always use `ConfigureAwait(false)` in these kind of libraries. But this method will introduce state-machine and await for something that's already completed. I know the implications of using `Task.FromResult(...)`, because any exceptions will be thrown immediately instead of via the task. Well-designed applications should be able to process that.